### DISCUSSION TITLE
How to use jk as Esc in VS Code with Neovim extension when it works in Neovim but not in VS Code?

### DISCUSSION THREAD

**Discussion by Firewallrbn** [vscode-neovim/vscode-neovim#2297] [Q&A]
I have the jk shortcut set up in Neovim to exit insert mode and return to normal mode with the following mapping in my init.vim:
inoremap jk <Esc
This works perfectly in Neovim: when I press jk quickly, it acts as Esc, and I can still use j and k individually without any issues.

However, when I try to replicate this behavior in VS Code using the Neovim extension, I encounter problems. Specifically, VS Code waits for the second key of the chord when I press j, which prevents me from typing j normally in insert mode.

I tried configuring this in keybindings.json like this:
`{
    "command": "vscode-neovim.send",
    "key": "j k",
    "when": "editorTextFocus && neovim.mode == 'insert'",
    "args": "<Esc>"
}`
But this causes the issue where VS Code waits for the second key (k), making it difficult to type j normally in insert mode.

Additionally, I tried the solution using jj mentioned in the README, but it didn't work either. I received an error in VS Code saying expected "array" when I tried to implement it, which seems to indicate a problem with the JSON structure.

**Comment by gaoqiangks** [comment]
The doc says,  add the following to your `settings.json` (attention, not `keyboard.json`)

```

    "vscode-neovim.compositeKeys": {
        "jj": {
            "command": "vscode-neovim.escape",
        },
    },
```

**Reply by vbfischer** [reply]
I do have this (in `settings.json`):

```json
  "vscode-neovim.compositeKeys": {
    "jj": {
        "command": "vscode-neovim.escape",
    },
  },
```

but it keeps giving me the warning 

> THe command "CompositeEscape1" and "CompositeEscape2" are deprecated

**Comment by little-yzp** [comment]
i have a problem，Can the plugin set the jk key instead of esc by setting init.lua or init.vim, or can it only be set by settings.json?

**Reply by xiyaowong** [reply]
In insert mode, the input is fully controlled by VSCode. You can add a keybinding to execute vscode-neovim.send or vscode-neovim.lua to perform the desired actions.

**Comment by giba0** [comment]
I don't know if you still have the problem, but in my case this worked (in settings.json):
```json
  "vscode-neovim.compositeKeys": {
    "jk": {
      "command": "vscode-neovim.lua",
      "args": [
        "vim.api.nvim_input('<ESC>')" 
      ]
    },
  }
```

**Reply by FallingSakura** [reply]
It works, ty!

**Reply by FallingSakura** [reply]
```
  "vscode-neovim.compositeKeys": {
    "jk": {
      "command": "vscode-neovim.escape",
    },
  },
```
This will be better
